taqueria-vscode-extension/src/lib/helpers.ts

```diff
@@ -34,15 +35,61 @@ export const COMMAND_PREFIX = 'taqueria.';
 const minNodeVersion = '16.13.1';
 
 export enum Commands {
-	init = 'taqueria.init',
-	scaffold = 'taqueria.scaffold',
-	install = 'taqueria.install',
-	uninstall = 'taqueria.uninstall',
-	optIn = 'taqueria.optIn',
-	optOut = 'taqueria.optOut',
-	originate = 'taqueria.originate',
+	init = 'init',
+	scaffold = 'scaffold',
+	install = 'install',
+	uninstall = 'uninstall',
+	optIn = 'optIn',
+	optOut = 'optOut',
+	originate = 'originate',
 }
 
+export type SmartContractLanguageInfo = {
+	fileExtensions: string[];
+	pluginName: string;
+	compilerName: SmartContractCompiler;
+};
+
+export type SmartContractCompiler = 'ligo' | 'archetype' | 'smartpy';
+
+export const smartContractLanguages: SmartContractLanguageInfo[] = [
+	{
+		fileExtensions: ['.mligo'],
+		pluginName: '@taqueria/plugin-ligo',
+		compilerName: 'ligo',
+	},
+	{
+		fileExtensions: ['.ligo'],
+		pluginName: '@taqueria/plugin-ligo',
+		compilerName: 'ligo',
+	},
+	{
+		fileExtensions: ['.religo'],
+		pluginName: '@taqueria/plugin-ligo',
+		compilerName: 'ligo',
+	},
+	{
+		fileExtensions: ['.jsligo'],
+		pluginName: '@taqueria/plugin-ligo',
+		compilerName: 'ligo',
+	},
+	{
+		fileExtensions: ['.py'],
+		pluginName: '@taqueria/plugin-smartpy',
+		compilerName: 'smartpy',
+	},
+	{
+		fileExtensions: ['.ts'],
```

Comment:
We don't support Typescript in SmartPy yet btw. Only Python

Reply:
@hu3man So should I not enable it when the user has a `.ts` file open? For instance?

Reply:
Yes, that would be ideal behaviour! 

May want to leave a comment there that we may need to refactor if and when we support TS via SP